runners/direct-java/src/main/java/org/apache/beam/runners/direct/EvaluationContext.java

```diff
@@ -158,7 +159,7 @@ public CommittedResult handleResult(
       TransformResult result) {
     Iterable<? extends CommittedBundle<?>> committedBundles =
         commitBundles(result.getOutputBundles());
-    if (result.getLogicalMetricUpdates() != null) {
+    if (!result.getLogicalMetricUpdates().isEmpty()) {
       metrics.commitLogical(completedBundle, result.getLogicalMetricUpdates());
```

Comment:
Comment on why we still need/want the if statement. My understanding is that we want it because otherwise we'd spin for no reason if we have contention, but we'd still be correct.